Updated Documentation + Github Action

tekrajchhetri opened the pull request:
The following are the changes made:
1. Updated the documentation, such as the details about the rapid release and list of libraries.
2. Fixed GitHub Action deployment issue caused by the deprecation of V3 on January 30.

tekrajchhetri committed:
added codespell

tekrajchhetri committed:
corrected spelling errors by Codespell

tekrajchhetri committed:
skip list added

tekrajchhetri committed:
list of libraries updated and publications added.

tekrajchhetri committed:
new rapid release related images added

tekrajchhetri committed:
Note added regarding known issues.

tekrajchhetri committed:
images updated + new data release page image added

tekrajchhetri committed:
manually added  "sphinxcontrib.mermaid" to enable mermaid diagram

tekrajchhetri committed:
readme updated to include 'sphinxcontrib-mermaid' information.

tekrajchhetri committed:
UI overview updated + rapid release information added.

tekrajchhetri committed:
updated config file for "AttributeError: 'list' object has no attribute 'items'"

tekrajchhetri committed:
"sphinxcontrib-mermaid" added.

tekrajchhetri committed:
phinxcontrib-mermaid added directly on yml file.

tekrajchhetri committed:
Update build_deploy.yml

tekrajchhetri committed:
list of libraries updated.

tekrajchhetri committed:
Updated documentation - list of libraries information

tekrajchhetri committed:
updated version because v3 has been deprecated as of Jan 30 2025

tekrajchhetri committed:
Update build_deploy.yml

tekrajchhetri committed:
v5.4.0 not found

tekrajchhetri committed:
Error - No uploaded artifact was found!

tekrajchhetri committed:
reverted changes and updated to actions/cache to v4

tekrajchhetri committed:
updated deploy-page to actions/deploy-pages@v4

tekrajchhetri committed:
link updated + assertion-evidence survey information added.

tekrajchhetri committed:
missing model information added

tekrajchhetri committed:
EviSense information added

tekrajchhetri committed:
Update list_of_libaries.md

tekrajchhetri committed:
Update list_of_libaries.md

kabilar commented on .github/workflows/build_deploy.yml:
```diff
@@ -32,14 +32,15 @@ jobs:
       run: |
-        pip install -r docs/requirements.txt
+        pip install -r docs/requirements.txt 
+        pip install sphinxcontrib-mermaid
```
Since this package is included in `docs/requirements.txt`, perhaps we can remove this line?

```suggestion
```

kabilar commented on README.md:
````diff
@@ -6,7 +6,7 @@ This repository contains the Jupyter Book documentation for BrainKB.
 
 ```
 pip install -U jupyter-book sphinx-autobuild
````
```suggestion
pip install -r docs/requirements.txt
```

kabilar replied:
Do we need to add `sphinx-autobuild` to `docs/requirements.txt`?

kabilar commented on README.md:
````diff
@@ -6,7 +6,7 @@ This repository contains the Jupyter Book documentation for BrainKB.
 ```
 pip install -U jupyter-book sphinx-autobuild
-
+pip install sphinxcontrib-mermaid
````
```suggestion
```

kabilar commented on docs/changes_log.md:
```diff
@@ -0,0 +1,17 @@
+  - Change Title
+  - GitHub
+  - Short Description
+* - date
+  - Ingest service
+  - [https://github.com/sensein/BrainKB/tree/ingestion-fapi-skeleton/ingestion_service](https://github.com/sensein/BrainKB/tree/ingestion-fapi-skeleton/ingestion_service)
+  - s
+  - git
```
Is this placeholder information?

tekrajchhetri replied:
That page is no longer relevant and removed.

kabilar commented on docs/list_of_libaries.md:
````diff
@@ -9,11 +9,40 @@
 ```
 
+## Ontologies/Schemas
+```{list-table} Ontologies/Schemas Developed as Part of BrainKB
````
Perhaps the subheading is a bit redundant?

![image](https://github.com/user-attachments/assets/c9cf1143-4af3-4419-8aa0-da454209c696)

```suggestion
```{list-table}
```

kabilar commented on docs/list_of_libaries.md:
````diff
@@ -9,11 +9,40 @@
+
+## Supporting Software and Libraries
 
 ```{list-table} Additional Software and Libraries Developed for BrainKB
````
```suggestion
```{list-table}
```

kabilar commented on docs/brainkbui.md:
````diff
@@ -1,5 +1,61 @@
+
+It is important to highlight that the data in AWS is organized hierarchically by release, adhering to a structured directory pattern that must be followed when uploading data. {numref}`brainkb_data_release_aws_s3_directory_structure` illustrates the organizational conventions used in AWS S3, with folders systematically arranged to represent releases and their subcategories. Similarly, {numref}`br ...
+
+```{mermaid} 
+sequenceDiagram
+    actor User
+    participant UI as User Interface
+    participant DB as Graph Database
+    participant S3 as AWS S3 Bucket
+
+    rect rgba(230, 210, 231, 0.5)
+    note right of User: Manual Steps
+    User ->> DB: Fetch Data from Graph Database
+    DB -->> User: Data Retrieved
+    User ->> S3: Upload Data to S3
+    S3 -->> User: Confirm Upload
+    end
+
+    rect rgba(200, 255, 200, 0.5)
+    note right of UI: Automated Steps
+    UI ->> S3: Fetch Data from S3
+    S3 -->> UI: Data Retrieved
+    UI ->> UI: Process and Visualize Data
+    UI ->> User: Display Data for Download
+    end
+```
````
The mermaid diagram doesn't seem to be displaying here: https://sensein.group/brainkbdocs/brainkbui.html

tekrajchhetri replied:
I will look into it. Didn't notice as I was able to see it locally.

![image](https://github.com/user-attachments/assets/266f50aa-6ee4-4769-b38c-c619e9a94765)

tekrajchhetri replied:
the new deployment has fixed this issue.

![image](https://github.com/user-attachments/assets/e84ec670-a262-4d09-a446-3ca8d612083a)

kabilar commented on docs/brainkbui.md:
```diff
@@ -1,5 +1,61 @@
+The sequence diagram below illustrates the steps involved in the rapid release process. Currently, the workflow is semi-automated, particularly in the stages of fetching data from the graph database (Oxigraph) and uploading it to an AWS S3 bucket.
+The data from AWS S3 is automatically retrieved, processed, and displayed in the UI, allowing users to download it seamlessly. {numref}`brainkb_data_release` presents the visualization of the data retrieved from the AWS S3 bucket and displayed in the UI.
+
+It is important to highlight that the data in AWS is organized hierarchically by release, adhering to a structured directory pattern that must be followed when uploading data. {numref}`brainkb_data_release_aws_s3_directory_structure` illustrates the organizational conventions used in AWS S3, with folders systematically arranged to represent releases and their subcategories. Similarly, {numref}`br ...
```
```suggestion
Data in AWS is organized hierarchically by release, adhering to a structured directory pattern that must be followed when uploading data ({numref}`brainkb_data_release_aws_s3_directory_structure`). {numref}`brainkb_data_release_aws_s3_ui_mapping` demonstrates how this AWS directory structure is mapped to the user interface.
```

kabilar commented on docs/images/brainkb_rapid_release.png:
Perhaps we could replace this diagram with the following, which can fit inline with the text better and will make edits in the future easier.
```
data_release
    └── release name
        └── date
            ├── individual
            └── combined
```

tekrajchhetri replied:
For now I will keep the current version, will change in the upcoming updates.

kabilar reviewed (changes requested):
Thank you @tekrajchhetri.  A few minor suggestions above.

kabilar commented:
> The following are the changes made:
> 
> 1. Updated the documentation, such as the details about the rapid release and list of libraries.
> 2. Fixed GitHub Action deployment issue caused by the deprecation of V3 on January 30.

Hi @tekrajchhetri, one suggestion for the future.  Perhaps we could submit changes to different components as separate pull requests. This way reviews on one component don't hold up the merge of another component.  Thanks for considering.

tekrajchhetri committed:
Update .github/workflows/build_deploy.yml

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>

tekrajchhetri committed:
Suggestion applied, moved installation of jupyter-book sphinx-autobuild to requirements

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>

tekrajchhetri committed:
Update README.md

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>

tekrajchhetri committed:
updated based on review comments

tekrajchhetri committed:
Merge branch 'docs' of github.com:sensein/brainkbdocs into docs

tekrajchhetri committed:
Update docs/list_of_libaries.md

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>

tekrajchhetri committed:
Merge branch 'docs' of github.com:sensein/brainkbdocs into docs

tekrajchhetri committed:
Delete changes_log.md

tekrajchhetri committed:
Update docs/brainkbui.md

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>

tekrajchhetri commented:
> > The following are the changes made:
> > 
> > 1. Updated the documentation, such as the details about the rapid release and list of libraries.
> > 2. Fixed GitHub Action deployment issue caused by the deprecation of V3 on January 30.
> 
> Hi @tekrajchhetri, one suggestion for the future. Perhaps we could submit changes to different components as separate pull requests. This way reviews on one component don't hold up the merge of another component. Thanks for considering.

@kabilar Sure. The reason why you see it like this way is because I update all the documentation things once and make the pull request accordingly. I think you're correct, keeping things separate will help in review as well.

tekrajchhetri committed:
Git action to deploy to AWS

tekrajchhetri committed:
Merge branch 'docs' of github.com:sensein/brainkbdocs into docs

tekrajchhetri committed:
debug statement added.

tekrajchhetri committed:
directory creation statement added.

tekrajchhetri committed:
Update deploy_aws.yml

tekrajchhetri committed:
updated config file  to include sphinxcontrib.mermaid  + workflow for SSH

tekrajchhetri committed:
Merge branch 'docs' of github.com:sensein/brainkbdocs into docs

tekrajchhetri committed:
Update deploy_aws.yml

tekrajchhetri committed:
apache installation

tekrajchhetri committed:
debug statement added to see if SSH is successful.

tekrajchhetri committed:
Updated ec2 users + host reading

tekrajchhetri committed:
VirtualHost 8040 configuration

tekrajchhetri committed:
fix for job for apache2.service failed because the control process exited with error code.

tekrajchhetri committed:
changed to docker deployment

tekrajchhetri committed:
updated docker file for permission issue

tekrajchhetri committed:
removed deploy on github pages

tekrajchhetri committed:
updated docker compose and docker file for deployment

tekrajchhetri committed:
Update docs/list_of_libaries.md

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>

tekrajchhetri commented:
> Thank you @tekrajchhetri. A few minor suggestions above.

Addressed the requested changes. Also, removed the git workflows for github pages deployment.

tekrajchhetri commented:
@kabilar I have addressed your comments and migrated to https://docs.brainkb.org/ as you've suggested.

kabilar commented on .github/workflows/deploy_aws.yml:
Hi @tekrajchhetri, curious as to what might be the benefits of deploying to an AWS EC2 as opposed to GitHub pages?  I would lean towards deploying to GitHub pages since GitHub manages the instance.

kabilar replied:
Here is an [example deployment](https://github.com/AllenInstitute/openscope_databook/blob/main/.github/workflows/deploy.yml) of Jupyter Book to GitHub pages.

tekrajchhetri replied:
@kabilar earlier, I was deploying using the GitHub pages and switched to AWS because of the sub-domain (docs.brainkb.org) as I was unsure about custom domain mapping in GitHub pages. Do you have any pointers for this?

kabilar replied:
Ah yes, GitHub will be able to handle the routes.  You'll just need the following:

1. CNAME file in this repository with `docs.brainkb.org`
2. Add a record in AWS Route 53 (and maybe we need to start using terraform to manage the infrastructure?):
    Record name: `docs.brainkb.org`
    Type: `CNAME`
    Routing policy: `Simple`
    Value: `sensein.github.io.`

For reference, see the DANDI Docs [terraform configuration](https://github.com/dandi/dandi-infrastructure/blob/724b235142f68103dd9108d9964b4890308da2ee/terraform/domain.tf#L47-L53) and [CNAME file](https://github.com/dandi/dandi-docs/blob/master/docs/CNAME).

tekrajchhetri replied:
@kabilar Thanks. I have updated CNAME to point to `docs.brainkb.org` and is now working fine.

kabilar commented on Dockerfile:
```diff
@@ -0,0 +1,17 @@
+WORKDIR /docs
+
+# We won't COPY here, because we rely on the volume mount for local dev.
+
```
```suggestion
# The container will always run in /docs
WORKDIR /docs

# We won't COPY here, because we rely on the volume mount for local dev.

# Install dependencies
RUN pip install --no-cache-dir -r docs/requirements.txt

```

tekrajchhetri committed:
Update Dockerfile

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>

tekrajchhetri committed:
removed aws deployment and restored github pages-based deployment

tekrajchhetri committed:
Merge branch 'docs' of github.com:sensein/brainkbdocs into docs

tekrajchhetri commented:
@kabilar As all the issues have been addressed, I will merge unless further comment. I have kept the `Dockerfile` and `docker-compose.yml` as it might be helpful if one wishes to deploy using docker such as for local testing.

kabilar reviewed (approved):
Thanks @tekrajchhetri.
